pycheribuild/projects/cross/cheribsd.py

```diff
@@ -68,6 +72,345 @@ def _clear_dangerous_make_env_vars():
+    def get_kabi_name(self, platform, kABI):
+        if kABI == KernelABI.NOCHERI:
+            return ""
+        elif kABI == KernelABI.HYBRID:
+            return "CHERI"
+        elif kABI == KernelABI.PURECAP:
+            return "CHERI{sep}PURECAP".format(sep=self.separator)
+
+    def get_platform_name(self, platform):
+        if type(platform) == set:
+            # Only use the first matching platform in the set
+            for key, name in self.platform_name_map.items():
+                if key in platform:
+                    return name
+            assert False, "Should not be reached..."
+        return self.platform_name_map[platform]
+
+    def get_flag_names(self, mfsroot=False, fuzzing=False):
+        flags = []
+        if mfsroot:
+            flags.append("MFS{sep}ROOT".format(sep=self.separator))
+        if fuzzing:
+            flags.append("FUZZ")
+        return flags
+
+    def _prepare_kernconf_context(self, platform, kABI, base_context=None, **kwargs):
+        if base_context is None:
+            base_context = self.kernconf_components
+        ctx = OrderedDict(base_context)
+        if "kabi_name" in ctx:
+            ctx["kabi_name"] = self.get_kabi_name(platform, kABI)
+        if "platform_name" in ctx:
+            ctx["platform_name"] = self.get_platform_name(platform)
+        if "flags" in ctx:
+            ctx["flags"] = self.separator.join(self.get_flag_names(**kwargs))
+        return ctx
+
+    def make_config(self, platform, kABI, base_context=None, **kwargs):
+        kernconf_ctx = self._prepare_kernconf_context(platform, kABI, base_context=base_context, **kwargs)
+
+        valid_ctx_items = (v for v in kernconf_ctx.values() if v is not None and v != "")
+        kernconf = self.separator.join(valid_ctx_items)
+        return CheriBSDConfig(kernconf, platform, kABI=kABI, **kwargs)
+
+
+class MIPSKernelConfigFactory(KernelConfigFactory):
+    platform_name_map = {
+        ConfigPlatform.QEMU: "MALTA64",
+        ConfigPlatform.BERI: "DE4"
+    }
+
+    def get_kabi_name(self, platform, kABI):
+        if platform == ConfigPlatform.BERI and kABI == KernelABI.NOCHERI:
+            return "BERI"
+        return super().get_kabi_name(platform, kABI)
+
+    def get_flag_names(self, usbroot=False, nfsroot=False, default=False, caprevoke=False,
+                       mfsroot=False, debug=False, benchmark=False, fuzzing=False, fett=False):
+        flags = []
+        if usbroot:
+            flags.append("USBROOT")
+        if nfsroot:
+            flags.append("NFSROOT")
+        flags += super().get_flag_names(mfsroot=mfsroot, fuzzing=fuzzing)
+        if benchmark:
+            flags.append("BENCHMARK")
+        return flags
+
+    def make_config(self, platform, kABI, base_context=None, **kwargs):
+        kernconf_ctx = self._prepare_kernconf_context(platform, kABI, base_context=base_context, **kwargs)
+
+        valid_ctx_items = (v for v in kernconf_ctx.values() if v is not None and v != "")
```

Comment:
If we return None for NOCHERI's kABI name can this lose the `!= ""`? Logically `""` would mean `-GFE` not `GFE`, for example.

Reply:
Yes, fixed now.